### ISSUE TITLE
Support additional client types

### ISSUE THREAD

**Issue by mchugh19** [SUSE/salt-netapi-client#84]
Salt's api supports additional methods of interaction. Specially, it would be helpful to support local_batch and runner in addition to the existing local client type. 

local_batch would allow for using the batch interface, while runner would allow for kicking off orchestrate jobs. 

http://docs.saltstack.com/en/latest/topics/netapi/index.html#client-interfaces

**Comment by lucidd** [comment]
The [run](http://suse.github.io/saltstack-netapi-client-java/docs/v0.2.0/com/suse/saltstack/netapi/client/SaltStackClient.html#run-java.lang.String-java.lang.String-com.suse.saltstack.netapi.AuthModule-java.lang.String-java.lang.String-java.lang.String-java.util.List-java.util.Map-) method already has a client parameter is that what you are looking for?

**Comment by mchugh19** [comment]
I don't think run returns a jid, just the minions responses that have returned. Also, we would need to specify a batch key with local_batch operations.

**Comment by lucidd** [comment]
I'm not sure i understand. Could you point me to the netapi function [here](http://docs.saltstack.com/en/latest/ref/netapi/all/salt.netapi.rest_cherrypy.html) and how you would call it to get what you want? A `curl` example would be nice. Same goes for the other 2 issues you opened with an example it would be easier to understand what exactly it is you want.

**Comment by mchugh19** [comment]
Sure can!

Here's an example a json post of the standard local client, using a flat "list" of minion IDs. 
```json
[{
    "arg": ["thing1", "thing2"],
    "client": "local",
    "expr_form": "list",
    "fun": "test.arg",
    "kwarg": {
        "test": "True"
    },
    "tgt": "webadmin.cws.dpdcn.unt.edu"
}]
```

Here's an example of a json request to kick off a local_batch client request. Note the additional batch key:value pair. 
```json
 [{
    "arg": ["thing1", "thing2"],
    "batch": "50%",
    "client": "local_batch",
    "expr_form": "glob",
    "fun": "test.arg",
    "kwarg": {
        "test": "True"
    },
    "tgt": "dyno2*"
}]
```

Finally, here's an example of a json request using the runner client and the orchestrate function
```json
[{
    "client": "runner",
    "expr_form": "glob",
    "fun": "state.orchestrate",
    "mods": "orchestration.refresh-apache",
    "pillar": {
        "target": "dyno2v8"
    },
    "tgt": "dynocache2v2*"
}]
```

In summary, it appears you will need to explicitly support local_batch since it requires the additional batch key as well as the runner client which requires the mods key. Supporting a pillar key would also be handy as shown in the runner example.

**Comment by lucidd** [comment]
Wheel and Runner functions are now supported both synchronously and asynchronously with #94. `local_batch` is not yet supported but hopefully soon.

**Comment by mchugh19** [comment]
Thanks all! Support for local_batch is necessary for the [jenkins saltstack plugin](/jenkinsci/saltstack-plugin) to make use of this library.

**Comment by lucidd** [comment]
Hey it should not be too hard to add `local_batch` support. It would only require an additional parameter for batch to the `callSync` methods in `LocalCall` and then passing that down with different client value. If you like you could try that out and report back how it works. I have not yet used local_batch so i don't no the implications of that.

**Comment by renner** [comment]
@mchugh19: We just merged #171 adding in support for the `local_batch` client. It would be great if you tested it with your jenkins plugin and let us know if it works. Just build a jar from the current `master` branch using `mvn package`.

The only unsupported clients are now `local_subset` and `ssh_async` which seems to be actually unimplemented in salt.

**Comment by mchugh19** [comment]
Fantastic! I've just relocated to Ireland from the US, so it will take me a little bit to get set up, but I'll get on it.
